### PR TITLE
Add Vultr provider

### PR DESCRIPTION
**Pull request by denego**
Provider - https://vultr.com/
Vultr API - https://www.vultr.com/api/
Terraform API - https://www.terraform.io/docs/providers/vultr/index.html

### CODE REVIEW

File: provider/vultr/main.tf [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+output "hostnames" {
+  value = vultr_server.host.*.hostname
+}
+
+output "private_ips" {
+  value = vultr_server.host.*.main_ip
+}
+
+output "private_network_interface" {
+  value = "$(ifconfig -s | grep -E '^ens\\w+' | cut -d' ' -f1)"
```
Comment (pstadler) [inline provider/vultr/main.tf:84]: oh boy, this is rather annoying.

File: variables.tf [hunk 2]
```diff
@@ -115,6 +115,65 @@ variable "packet_user_data" {
...
+
+variable "vultr_ssh_key_ids" {
+  /*
+    curl -H "API-Key: $KEY" "https://api.vultr.com/v1/sshkey/list" \
+      | jq '.[].SSHKEYID'
+  */
+  default = [""]
+}
+
+variable "vultr_rate_limit" {
```
Comment (pstadler) [inline variables.tf:156]: oh boy, this is very annoying.

File: variables.tf [hunk 3]
```diff
@@ -115,6 +115,65 @@ variable "packet_user_data" {
   default = ""
 }
 
+/* vultr */
+
+variable "vultr_api_key" {
+  default = ""
+}
+
+variable "vultr_region_id" {
```
Comment (pstadler) [inline variables.tf:124]: isn't there a better way, or another terraform resource for fetching regions, plans, os, etc.?

Comment (denego) [reply variables.tf]: Ahh, you knew there is! Thanks for the lesson.

File: variables.tf (hunk 2 above)
Comment (denego) [reply variables.tf]: Let's hide this under the hood, so there is no vultr_rate_limit variable, as default is enough for 3 nodes. But let the module have the parameter for those who need to adjust it.

File: provider/vultr/main.tf (hunk 1 above)
Comment (denego) [reply provider/vultr/main.tf:122]: I can hardcode this to `ens3`, but I'm not sure the value is the same for all locations and won't change in the future.

File: variables.tf (hunk 3 above)
Comment (pstadler) [reply variables.tf]: yeah, I've seen this with other providers.

File: provider/vultr/main.tf (hunk 1 above)
Comment (pstadler) [reply provider/vultr/main.tf:122]: woah, how does this even work, is this command running on the remote per default?

Comment (denego) [reply provider/vultr/main.tf:122]: I've noticed the value is used only in the ufw module. The string inserted as-is into `ufw.sh` script making the script look like:
`ufw allow in on $(ifconfig -s | grep -E '^ens\\w+' | cut -d' ' -f1) to any port`.
And then the shell intepreter on the node do the work.

**Comment by pstadler** [comment]
stalled.

**Pull request closed without merging**